File: backend/src/main/java/com/example/backend/entities/DanhMucVaccine.java
package com.example.backend.entities;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "DANH_MUC_VACCINE")
@Data
public class DanhMucVaccine {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String ten_vaccine;
    private String phong_benh;
    private int ngay_tuoi;
    private String phuong_thuc;

    @OneToMany(mappedBy = "danhMucVaccine",cascade = CascadeType.ALL)
    @JsonIgnore
    private List<LichTiemThucTe> lichTiemThucTes;
}

File: smart_chicken_farming/lib/model/DanhMucVaccine.dart
class Danhmucvaccine {
  final int? id;
  final String? ten_vaccine;
  final int? ngay_tuoi;
  final String? phong_benh;
  final String? phuong_thuc;

  Danhmucvaccine({
    this.id,
    this.ten_vaccine,
    this.ngay_tuoi,
    this.phong_benh,
    this.phuong_thuc,
  });

  factory Danhmucvaccine.fromJson(Map<String, dynamic> json) {
    return Danhmucvaccine(
      id: json['id'] as int?,
      ten_vaccine: (json['ten_vaccine'] ?? json['tenVaccine']) as String?,
      ngay_tuoi: (json['ngay_tuoi'] ?? json['ngayTuoi']) as int?,
      phong_benh: (json['phong_benh'] ?? json['phongBenh']) as String?,
      phuong_thuc: (json['phuong_thuc'] ?? json['phuongThuc']) as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'ten_vaccine': ten_vaccine,
      'ngay_tuoi': ngay_tuoi,
      'phong_benh': phong_benh,
      'phuong_thuc': phuong_thuc,
    };
  }
}